src/gt4py/next/iterator/transforms/fuse_as_fieldop.py

```diff
@@ -119,9 +175,7 @@ def fuse_as_fieldop(
                 pass
             elif cpm.is_call_to(arg, "if_"):
```

Comment:
I was going through the whole file again and got stuck here. Can we briefly discuss
- if it makes the code cleaner, if instead of building an applied field_op we extract the stencil_body and args
- instead of special casing for `if` we support any function (because only `if` should be marked as eligible, but if we change the algorithm to mark eligible, we should not have to touch this line)